core/modules/date/date.install

```diff
@@ -101,3 +101,18 @@ function date_update_1001() {
   // Remove old system table entries for merged in modules.
   db_query("DELETE FROM {system} WHERE name IN ('date_api', 'date_popup', 'date_views') AND type = 'module'");
 }
+
+/**
+ * Remove entries with null values from date field tables.
+ */
+function date_update_1002() {
+  // Get all date fields.
+  $fields = field_info_fields();
+  foreach ($fields as $field_name => $field) {
+    if (in_array($field['type'], array('date', 'datetime', 'datestamp'))) {
+      db_delete('field_data_' . $field_name)
```

Comment:
We also need to consider `field_revision...`, otherwise we get inconsistencies between the two tables.

Reply:
Good catch! Will do.

Reply:
Done